src/language.h

```diff
@@ -99,21 +102,11 @@ class TextString {
 
 	void Clear();
 
-	/**
-	 * Get the string in the currently selected language.
-	 * @return Text of this string in the currently selected language.
-	 */
-	std::string GetString() const
-	{
-		if (_current_language < 0 || _current_language >= LANGUAGE_COUNT) return "<out of bounds>";
-		if (this->languages[_current_language] != nullptr) return this->languages[_current_language];
-		if (this->languages[LANG_EN_GB] != nullptr) return this->languages[LANG_EN_GB];
-		return "<no-text>";
-	}
+	std::string GetString(int64 amount) const;
 
 	/* Memory is not owned. */
-	const char* name;                       ///< Name of the string.
-	const char* languages[LANGUAGE_COUNT];  ///< The string in all languages.
+	const char* name;                                    ///< Name of the string.
+	std::vector<const char*> languages[LANGUAGE_COUNT];  ///< The string in all languages, indexed by plural form.
```

Comment:
How messy, different languages have different number of plural forms, and now you can't even push that problem down to the language.

You may want to turn this upside down.
Have proper `Language` that contains `TextPhrase`-es. (Basically "string" but that has too many confusing meanings.)
A TextPhrase may have multiple plural forms.

The current TextString (with a string in all languages) then doesn't exist anymore (although you can construct it again by making it point to all relevant TextPhrase objects), not sure how bad that is.

Lot of work, but I think it will make life more comfortable.